### ISSUE TITLE
🐛 Biome LSP: Incorrect errors from for `*.svelte` files without `<script>` tag, and config `extends` is not applying in monorepo

### ISSUE THREAD

**Issue by PleahMaCaka** [biomejs/biome#7127]
### Environment information
```
CLI:
  Version:                      2.1.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.19.4
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.19

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```
## Reproduction Steps

1. Clone https://github.com/PleahMaCaka/biome-report-minimal
2. Run `bunx biome check`. You’ll see that there’s no problem in the CLI.

- **Problem 1:** Open [`./packages/app/src/lib/routes/+layout.svelte`](./packages/app/src/lib/routes/+layout.svelte). The biome config in the app package is not applied. The config is [here](./packages/app/biome.json).

<img width="776" height="398" alt="Image" src="https://github.com/user-attachments/assets/06d72193-240e-4c46-bb24-4e4ee9bf5f59" />

- **Problem 2:** Open [`./packages/app/src/lib/routes/+page.svelte`](./packages/app/src/lib/routes/+page.svelte). There’s no problem by default, but if I edit any line, the Biome LSP shows strange behavior.

#### Related
Discord Thread: [help/Biome config is not working for svelte monorepo](https://discord.com/channels/1132231889290285117/1401855282220503102)
(Maybe) Related issue comment: https://github.com/biomejs/biome/issues/6434#issuecomment-2996874030

| Before Edit | After Edit |
|-------------|------------|
| <img width="243" height="120" alt="Image" src="https://github.com/user-attachments/assets/2ad82e35-427f-4aab-bb10-8d9ddee9a892" /> | <img width="950" height="167" alt="Image" src="https://github.com/user-attachments/assets/dbc18d7e-a7c9-4617-84ce-544e65204ed3" /> |

### Write More with TypeScript:

<img width="669" height="554" alt="Image" src="https://github.com/user-attachments/assets/8d38e39a-be29-4409-b827-eed401e68f83" />

Also, this is not an IDE-specific problem. See this:

| Zed | VSCode |
|-----------|-----------|
| <img width="805" height="790" alt="Image" src="https://github.com/user-attachments/assets/7add0035-8ff0-4190-9f56-f689c6f4b2ed" /> | <img width="871" height="700" alt="Image" src="https://github.com/user-attachments/assets/ee573215-1094-4f76-9778-9a1216c24f58" /> |

This is my server-side biome info:
```
* Server: biome (id 51)

* Binary: LanguageServerBinary {
    path: "/usr/bin/node",
    arguments: [
        "/home/pleahmacaka/CodeHere/biome-report-minimal/node_modules/@biomejs/biome/bin/biome",
        "lsp-proxy",
    ],
    env: Some(
        {
            ...some env vars
        },
    ),
}

* Registered workspace folders:
/home/pleahmacaka/CodeHere/biome-report-minimal

* Capabilities: {
  "positionEncoding": "utf-16",
  "textDocumentSync": 2,
  "codeActionProvider": {
    "codeActionKinds": [
      "quickfix.biome",
      "quickfix.suppressRule.topLevel.biome",
      "quickfix.suppressRule.inline.biome",
      "source.organizeImports.biome",
      "source.fixAll.biome",
      "refactor.biome",
      "refactor.extract.biome",
      "refactor.inline.biome",
      "refactor.rewrite.biome",
      "source.biome"
    ]
  },
  "documentFormattingProvider": true,
  "documentRangeFormattingProvider": true,
  "documentOnTypeFormattingProvider": {
    "firstTriggerCharacter": "}",
    "moreTriggerCharacter": [
      "]",
      ")"
    ]
  },
  "workspace": {
    "workspaceFolders": {
      "supported": true,
      "changeNotifications": true
    ]
  }
}

* Configuration: {
  "biome": {
    "configurationPath": "/home/pleahmacaka/CodeHere/biome-report-minimal/biome.json",
    "requireConfiguration": true
  }
}
```

### Expected result

Everything should just work.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by yonben-lendbuzz** [comment]
Might be similar but I have a nested config in my monorepo as well. It adds a warning on restricted imports. It works on CLI but in the IDE I get no warning.

**Comment by dyc3** [comment]
I don't believe this problem is necessarily related to monorepos. The parsing, as it stands right now, should only be constrained to the `script` tag (svelte is currently only partially supported). I've seen this behavior when I create new svelte files with my IDE sometimes.

The easiest workaround right now is to restart biome via your editor.

**Comment by pleahmacaka** [comment]
> I don't believe this problem is necessarily related to monorepos. The parsing, as it stands right now, should only be constrained to the `script` tag (svelte is currently only partially supported). I've seen this behavior when I create new svelte files with my IDE sometimes.
> 
> The easiest workaround right now is to restart biome via your editor.


But it works when I uses as a single configuration. Of course, I tried restarting the LSP and my IDE multiple times, but simply restarting did not help.

Just 
1. Copy the [`override` property](https://github.com/PleahMaCaka/biome-report-minimal/blob/main/packages/app/biome.json) in `packages/app/biome.json` into [`./biome.json`](https://github.com/PleahMaCaka/biome-report-minimal/blob/main/biome.json) 
2. Remove `packages/app/biome.json`
3. Restart LSP -> it works! (Edit: When `<script>` tag in `*.svelte` file. See below.)

**Comment by pleahmacaka** [comment]
The issues have been somewhat clarified.

1. The `(biome parse)` errors appearing in various `*.svelte` files occur when there is no `<script>` tag present. This can be reproduced as follows:
   - `git clone https://github.com/pleahmacaka/biome-report-minimal`
   - `cd packages/app` -> `bun dev` to generate `.svelte-kit` files and then shut down (Ctrl + C).
   - Edit `packages/app/src/route/+page.svelte` to see the weird errors.
   - Add a script tag as shown below and restart the LSP (an empty script tag works too):
     ```html
     <script lang="ts">
     </script>
     ```
   - No errors.

2. However, it still seems unable to extend the `./biome.json` settings. My root biome configuration includes the following:
   ```json
   "javascript": {
     "formatter": {
       ...
       "semicolons": "asNeeded"
     }
   }
   ```

   And the app package has these settings:
   ```json
   "overrides": [
     {
       "includes": ["**/*.svelte", "**/*.astro", "**/*.vue"],
       "linter": {
         "rules": {
           "style": {
             "useConst": "off",
             "useImportType": "off"
           },
           "correctness": {
             "noUnusedVariables": "off",
             "noUnusedImports": "off"
           }
         }
       }
     }
   ]
   ```

But when formatting, none of these are actually applied.

https://github.com/user-attachments/assets/fa18e728-69e8-4166-a41e-0b900dcf0550

The issue with extends not working in biome.json seems to be temporarily solvable by using only the root configuration. However, cases without <script> tags seem common. Is this problem occurring only for me?

**Comment by novaotp** [comment]
> Is this problem occurring only for me?

Nope, I'm having the same issues when I create a new file and add markup without any script tags.

My workaround is to add an empty script tag, save the file, close it and re-open it. It's a bit annoying, but I don't mind it too much since I always need a script tag anyway.

**Comment by pleahmacaka** [comment]
> > Is this problem occurring only for me?
> 
> Nope, I'm having the same issues when I create a new file and add markup without any script tags.
> 
> My workaround is to add an empty script tag, save the file, close it and re-open it. It's a bit annoying, but I don't mind it too much since I always need a script tag anyway.

Thank you for sharing your experience.

I temporarily resolve this issue using the same method as yours. However, due to the same cause, there is a hassle of restarting the LSP every time to create components. The pages and layout don't bother much, but it's quite annoying when work with new project.

**Comment by glass-ships** [comment]
I'm having the same issue with Vue files too, actually... does biome not parse the <template> section of vue files? this may be expected, but does not seem ideal.   


<img width="514" height="326" alt="Image" src="https://github.com/user-attachments/assets/4a72a4af-e84f-4289-a88a-a7bbec495045" />

**Comment by dyc3** [comment]
Biome can't currently evaluate rules that cross language boundaries. 

See: https://biomejs.dev/internals/language-support/

**Comment by pleahmacaka** [comment]
Fixed as of 2.5.7 for me, closing.
